shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-route/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/dml/impl/ShardingInsertStatementValidator.java

```diff
@@ -62,6 +78,9 @@ public void preValidate(final ShardingRule shardingRule, final SQLStatementConte
         if (insertSelectSegment.isPresent() && !isAllSameTables(tablesContext.getTableNames()) && !shardingRule.isAllBindingTables(tablesContext.getTableNames())) {
             throw new ShardingSphereException("The table inserted and the table selected must be the same or bind tables.");
         }
+        if (insertSelectSegment.isPresent() && isNeedMergeShardingValues(sqlStatementContext, shardingRule)) {
+            checkSubqueryShardingValues(shardingRule, sqlStatementContext, parameters, schema);
```

Comment:
Have you forgotten to remove the `checkSubqueryShardingValues` method from the `ShardingSQLRouter` class? 😉

Reply:
OK